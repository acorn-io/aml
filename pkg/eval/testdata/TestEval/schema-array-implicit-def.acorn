define Foo: {
    foo: ["a", "b"]
    bar: ["a", "b"]
    baz: ["a", "b"]
}

Foo({
    foo: []
    baz: ["x", "y"]
})
