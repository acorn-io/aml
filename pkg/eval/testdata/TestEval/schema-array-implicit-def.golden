{
  "Foo": {
    "type": "object",
    "properties": {
      "bar": {
        "$ref": "#/$defs/Foo.bar"
      },
      "baz": {
        "$ref": "#/$defs/Foo.baz"
      },
      "foo": {
        "$ref": "#/$defs/Foo.foo"
      }
    },
    "defs": {
      "Foo.bar": {
        "type": "array",
        "items": [
          {
            "$ref": "#/$defs/Foo.bar[0]"
          }
        ]
      },
      "Foo.bar[0]": {
        "type": "string"
      },
      "Foo.baz": {
        "type": "array",
        "items": [
          {
            "$ref": "#/$defs/Foo.baz[0]"
          }
        ]
      },
      "Foo.baz[0]": {
        "type": "string"
      },
      "Foo.foo": {
        "type": "array",
        "items": [
          {
            "$ref": "#/$defs/Foo.foo[0]"
          }
        ]
      },
      "Foo.foo[0]": {
        "type": "string"
      }
    }
  },
  "bar": [
    "a",
    "b"
  ],
  "baz": [
    "x",
    "y"
  ],
  "foo": []
}